ivadomed/utils.py

```diff
@@ -277,6 +277,58 @@ def get_arguments(parser, args):
     return args
 
 
+def add_version_info(context):
+    path_data = context.get(ConfigKW.LOADER_PARAMETERS).get(LoaderParamsKW.PATH_DATA)
+
+    context[ConfigKW.IVADOMED_VERSION] = _version_string()
+    datasets_version = get_datasets_versions(context)
+    datasets_dict = {}
+
+    path_data = format_path_data(path_data)
+    for i_dataset in range(len(path_data)):
+        origin = get_origin(path_data[i_dataset])
+        if origin != "?!?":
+            datasets_dict[origin] = datasets_version[i_dataset]
+        else:
+            datasets_dict[path_data[i_dataset]] = "data_is_not_annexed"
+
+    context[ConfigKW.DATASETS_VERSION] = datasets_dict
+
+
+def get_datasets_versions(context):
+    path_data = context.get(ConfigKW.LOADER_PARAMETERS).get(LoaderParamsKW.PATH_DATA)
+    datasets_version = []
+
+    if isinstance(path_data, str):
+        datasets_version = [__get_commit(path_to_git_folder=path_data)]
+    elif isinstance(path_data, list):
+        for Dataset in path_data:
+            datasets_version.append(__get_commit(path_to_git_folder=Dataset))
+
+    return datasets_version
+
+
+def get_origin(path_to_git_folder):
+    """Get GIT remote origin.
+
+    Args:
+        path_to_git_folder (str): Path to GIT folder.
+    Returns:
+        str: git remote origin.
+    """
+    path_to_git_folder = Path(path_to_git_folder).expanduser().absolute()
+
+    p = subprocess.Popen(["git", "remote", "get-url", "origin"], stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                         cwd=path_to_git_folder)
+    output, _ = p.communicate()
+    status = p.returncode
+    if status == 0:
+        origin = output.decode().strip()
+    else:
```

Comment:
So if git is not in the system, I guess it will return `?!?` ? Could we also maybe inform the user a bit?

Reply:
I replace ?!? with directory name later in the code here https://github.com/ivadomed/ivadomed/pull/1119/files#diff-63d4df1cf0a430b8d02d7e2f2ec391316564414e7c827e89d7326d73efe67456R290